rosetta-source/src/main/java/org/isda/cdm/codelist/CodeListTransformer.java

```diff
@@ -0,0 +1,146 @@
+package org.isda.cdm.codelist;
+
+import cdm.base.staticdata.codelist.CodeList;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.regnosys.rosetta.common.serialisation.RosettaObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.xml.transform.*;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.transform.stream.StreamSource;
+import java.io.*;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.*;
+
+/**
+ * This class transforms `https://docs.oasis-open.org/codelist/ns/genericode/1.0/` codelist XML files to JSON-serialized CDM format using an XSLT script.
+ * The transformation applies to all XML files in a specified directory.
+ * Results are stored in the specified output directory.
+ */
+public class CodeListTransformer {
+
+    private static final Logger logger = LoggerFactory.getLogger(CodeListTransformer.class);
+
+    private static final ObjectMapper mapper = RosettaObjectMapper.getNewRosettaObjectMapper();
```

Comment:
`ObjectMapper`s tend to hold on to much memory, and when it is static, this memory will never be cleaned up until the Java process finishes. If possible, I would like to try and mitigate this.

Suggestion: make the mapper non-static, make all of the methods non-static except for main, and in your `main` run `new CodeListTransformer().transformDirectory` instead.